src/api/app/models/kiwi/image.rb

```diff
@@ -141,7 +141,37 @@ def outdated?
   end
 
   def default_package_group
-    package_groups.find_or_create_by(kiwi_type: :image, pattern_type: 'onlyRequired')
+    package_groups.find_or_create_by(kiwi_type: :image)
+  end
+
+  def find_binaries_by_name(query)
+    finder = /\A#{Regexp.quote(query)}/
+    binaries_available.select { |package, _| finder.match(package.to_s) }
+  end
+
+  def binaries_available
+    return {} unless package
+    Rails.cache.fetch("kiwi_image_binaries_available_#{md5_last_revision}", expires_in: 5.minutes) do
+      if use_project_repositories
+        Backend::Api::BuildResults::Binaries.available_in_project(package.project.name)
+      else
+        url_repositories = []
+        obs_repositories = []
+        repositories.each do |repository|
+          if repository.obs_source_path?
+            obs_repositories << repository.project_for_type_obs + '/' + repository.repository_for_type_obs
+          else
+            url_repositories << repository.source_path
+          end
+        end
+        binaries = Backend::Api::BuildResults::Binaries.available_in_urls(package.project.name, url_repositories)
+        obs_binaries = Backend::Api::BuildResults::Binaries.available_in_repositories(package.project.name, obs_repositories)
```

Comment:
Fetching that in one step is not possible?

Reply:
Actually it seems to work to combine the url & path parameters so you can do it one step and you don't need the next steps.